src/sage/modules/free_module_pseudomorphism.py

```diff
@@ -0,0 +1,443 @@
+            sage: Frob = Fq.frobenius_endomorphism()
+
+            sage: f = M.pseudohom([[1,1,1], [2,2,2], [3,3,3]], Frob)
+            sage: f  # indirect doctest
+            Free module pseudomorphism (twisted by z |--> z^7) defined by the matrix
+            [1 1 1]
+            [2 2 2]
+            [3 3 3]
+            Domain: Vector space of dimension 3 over Finite Field in z of size 7^3
+            Codomain: Vector space of dimension 3 over Finite Field in z of size 7^3
+
+            sage: g = M.pseudohom([[1,1,1], [2,2,2], [3,3,3]], Frob, side="right")
+            sage: g  # indirect doctest
+            Free module pseudomorphism (twisted by z |--> z^7) defined as left-multiplication by the matrix
+            [1 1 1]
+            [2 2 2]
+            [3 3 3]
+            Domain: Vector space of dimension 3 over Finite Field in z of size 7^3
+            Codomain: Vector space of dimension 3 over Finite Field in z of size 7^3
+        """
+        twist = self.parent()._ore._repr_twist()
+        s = "Free module pseudomorphism (%s) defined " % twist
+        if self._side == "right":
+            s += "as left-multiplication "
+        s += "by the matrix\n%s\n" % self.matrix()
+        s += "Domain: %s\n" % self.domain()
+        s += "Codomain: %s" % self.codomain()
+        return s
+
+    def matrix(self):
+        r"""
+        Return the underlying matrix of this pseudomorphism.
+
+        It is defined as the matrix `M` whose lines (resp. columns if
+        ``side`` is ``right``) are the coordinates of the images of
+        the distinguished basis of the domain.
+
+        EXAMPLES::
+
+            sage: Fq.<z> = GF(7^3)
+            sage: Frob = Fq.frobenius_endomorphism()
+            sage: M = Fq^3
+            sage: f = M.pseudohom([[1, z, 3], [0, 1, z^2], [z+1, 1, 1]], Frob)
+            sage: f.matrix()
+            [    1     z     3]
+            [    0     1   z^2]
+            [z + 1     1     1]
+
+            sage: e1, e2, e3 = M.basis()
+            sage: f(e1)
+            (1, z, 3)
+            sage: f(e2)
+            (0, 1, z^2)
+            sage: f(e3)
+            (z + 1, 1, 1)
+
+        TESTS::
+
+            sage: v = M.random_element()
+            sage: f(v) == vector([Frob(c) for c in v]) * f.matrix()
+            True
+        """
+        if self._side == "left":
+            return self._matrix.__copy__()
+        else:
+            return self._matrix.transpose()
+
+    def twisting_derivation(self):
+        r"""
+        Return the twisting derivation of the pseudomorphism
+        (or ``None`` if the twisting derivation is zero).
+
+        EXAMPLES::
+
+            sage: P.<x> = ZZ[]
+            sage: d = P.derivation()
+            sage: M = P^2
+            sage: f = M.pseudohom([[1, 2*x], [x, 1]], d)
+            sage: f.twisting_derivation()
+            d/dx
+
+        ::
+
+            sage: Fq.<z> = GF(7^3)
+            sage: Frob = Fq.frobenius_endomorphism()
```

Comment:
Yes. This is better.

By the way, the doc build seems to have a bug, as seen in https://doc-pr-38650--sagemath.netlify.app/html/en/reference/modules/sage/modules/free_module_pseudomorphism#sage.modules.free_module_pseudomorphism.FreeModulePseudoMorphism.twisting_derivation

You may ignore the bug here.

Reply:
Well, I should confess that I don't know how o fix this.